### PR TITLE
Bbmap bbnorm

### PR DESCRIPTION
**Pull request by danilodileo**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/metatdenovo/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/metatdenovo _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

Hi, I added bbnorm to the pipeline. It works and it can be selected instead of diginorm.

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -115,6 +115,25 @@ process {
         ]
     }
 
+    withName: BBMAP_BBNORM {
+        ext.args = [
+            params.target ? "target=${params.target}" : '',
+            params.min    ? "min=${params.min}" : '',
+            ].join(' ').trim()
```
Comment (erikrikarddaniel) [inline conf/modules.config]:
```suggestion
        ].join(' ').trim()
```

File: nextflow.config [hunk 2]
```diff
@@ -30,6 +30,9 @@ params {
     sequence_filter             = null
 
     // Digital normalization options
+    bbnorm                      = false
+    target                      = 100
+    min                         = 5
```
Comment (erikrikarddaniel) [inline nextflow.config]:
```suggestion
    bbnorm_target               = 100
    bbnorm_min                  = 5
```

Comment (erikrikarddaniel) [reply nextflow.config]: Better with more explicit names, I think.

File: nextflow_schema.json [hunk 3]
```diff
@@ -126,6 +126,21 @@
                     "type": "boolean",
                     "description": "Skip FastQC."
                 },
+                "bbnorm": {
+                    "type": "boolean",
+                    "description": "Perform normalization to reduce sequencing depth.",
+                    "help_text": "Normalization is performed following the example in https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbnorm-guide/"
+                },
+                "target": {
```
Comment (erikrikarddaniel) [inline nextflow_schema.json]:
```suggestion
                "bbnorm_target": {
```

File: nextflow_schema.json [hunk 4]
```diff
@@ -126,6 +126,21 @@
...
+                    "type": "integer",
+                    "default": 100,
+                    "description": "produce an output file of reads with an average depth of nx. "
+                },
+                "min": {
```
Comment (erikrikarddaniel) [inline nextflow_schema.json]:
```suggestion
                "bbnorm_min": {
```

File: conf/modules.config (hunk 1 above)
Comment (erikrikarddaniel) [inline conf/modules.config]:
```suggestion
        ext.args = [
            params.target ? "target=${params.bbnorm_target}" : '',
            params.min    ? "min=${params.bbnorm_min}" : '',
        ].join(' ').trim()
```

**Review by erikrikarddaniel — APPROVED** [review]
Great work! Only a tiny thing: I think it's better to add `bbnorm_` in front of the bbnorm params to make it clearer what they belong to.

File: conf/modules.config [hunk 5]
```diff
@@ -115,6 +115,25 @@ process {
         ]
     }
 
+    withName: BBMAP_BBNORM {
+        ext.args = [
+            params.bbnorm_target ? "target=${params.target}" : '',
```
Comment (danilodileo) [inline conf/modules.config]:
```suggestion
            params.bbnorm_target ? "target=${params.bbnorm_target}" : '',
```

File: conf/modules.config [hunk 6]
```diff
@@ -115,6 +115,25 @@ process {
...
+            params.bbnorm_min    ? "min=${params.min}" : '',
```
Comment (danilodileo) [inline conf/modules.config]:
```suggestion
            params.bbnorm_min    ? "min=${params.bbnorm_min}" : '',
```

**Pull request merged**